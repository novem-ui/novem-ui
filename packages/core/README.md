# @novem-ui/core

This lib exports all the Novem UI components