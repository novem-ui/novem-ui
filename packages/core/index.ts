export { default as NovemProvider } from './src/novem-provider'

export { default as Badge } from '@novem-ui/badge'
export * from '@novem-ui/badge'

export { default as Button, ButtonProps } from '@novem-ui/button'

export { default as Switch, SwitchProps } from '@novem-ui/switch'

export { Heading, Label, Paragraph, HeadingProps, LabelProps, ParagraphProps } from '@novem-ui/text'

export { Input, TextField, TextFieldProps } from '@novem-ui/input'

export { default as theme } from '@novem-ui/theme'
export type { Theme } from '@novem-ui/theme'
