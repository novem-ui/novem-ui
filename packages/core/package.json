{
  "name": "@novem-ui/core",
  "version": "0.0.1-alpha.1",
  "description": "Core components",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/core#readme",
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/core"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/core --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "dependencies": {
    "@novem-ui/badge": "^0.0.1-alpha.1",
    "@novem-ui/theme": "^0.0.1-alpha.1"
  },
  "devDependencies": {
    "@types/novem-theme": "^0.0.1-alpha.1",
    "@types/react": "^17.0.3"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.3.0",
    "polished": "^4.1.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
