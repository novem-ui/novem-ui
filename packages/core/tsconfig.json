{
  "extends": "../../tsconfig.json",
  "files": [],
  "include": ["**/*.ts", "**/*.tsx", "../../emotion.d.ts"],
  "exclude": ["node_modules/**", "dist/**"]
}
