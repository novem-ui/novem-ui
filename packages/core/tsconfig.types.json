{
  "extends": "../../tsconfig.json",
  "files": [],
  "compilerOptions": {
    "emitDeclarationOnly": true,
    "declaration": true,
    "declarationDir": "../../dist/",
    "declarationMap": true,
    "types": ["novem-theme"]
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules/**", "dist/**", "**/*.spec.tsx", "**/*.stories.tsx"]
}
