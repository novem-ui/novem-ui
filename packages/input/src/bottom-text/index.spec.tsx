import React from 'react'

import { render } from '@testing-library/react'
import { NovemProvider, theme } from '@novem-ui/core'
import { matchers } from '@emotion/jest'

import BottomText, { BotttomTextProps } from '.'

beforeAll(() => {
  expect.extend(matchers)
})

const setup = (props: BotttomTextProps = {}) => {
  const { container } = render(
    <NovemProvider>
      <BottomText {...props} />
    </NovemProvider>
  )

  const paragraph = container.querySelector('p')

  return { container, paragraph }
}

describe('input/bottom-text', () => {
  test('should render a helper', () => {
    const { container, paragraph } = setup({ helper: 'Helper text' })

    expect(container).toMatchSnapshot()
    expect(paragraph).toHaveStyleRule('color', theme.colors.palette.grey[400])
  })

  test('should render an error message (from string)', () => {
    const { container, paragraph } = setup({ error: 'Error text' })

    expect(container).toMatchSnapshot()
    expect(paragraph).toHaveStyleRule('color', theme.colors.palette.red[500])
  })

  test('should render an error message (from object)', () => {
    const { container, paragraph } = setup({ error: new Error('Error text') })

    expect(container).toMatchSnapshot()
    expect(paragraph).toHaveStyleRule('color', theme.colors.palette.red[500])
  })

  test('should render nothing', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })
})
