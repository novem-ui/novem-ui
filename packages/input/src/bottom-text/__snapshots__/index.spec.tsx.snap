// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input/bottom-text should render a helper 1`] = `
.emotion-0 {
  color: rgba(154,152,151,0.85);
  font-size: 12px;
  margin: 0;
  margin-left: 0.5rem;
  margin-top: 0.25rem;
}

<div>
  <p
    class="emotion-0"
    color="grey.400"
  >
    Helper text
  </p>
</div>
`;

exports[`input/bottom-text should render an error message (from object) 1`] = `
.emotion-0 {
  color: #EB5757;
  font-size: 12px;
  margin: 0;
  margin-left: 0.5rem;
  margin-top: 0.25rem;
}

<div>
  <p
    class="emotion-0"
    color="red.500"
  >
    Error text
  </p>
</div>
`;

exports[`input/bottom-text should render an error message (from string) 1`] = `
.emotion-0 {
  color: #EB5757;
  font-size: 12px;
  margin: 0;
  margin-left: 0.5rem;
  margin-top: 0.25rem;
}

<div>
  <p
    class="emotion-0"
    color="red.500"
  >
    Error text
  </p>
</div>
`;

exports[`input/bottom-text should render nothing 1`] = `<div />`;
